### ISSUE TITLE
[BUG] [Elm] String Type schemas cause the Data.elm to not be generated

### ISSUE THREAD

**Issue by alehed** [OpenAPITools/openapi-generator#10398]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The elm generator doesn't create a Data.elm in src/Api/ which leads to non-compiling code if the api spec contains a schema which is simply a string (see provided gist).

##### openapi-generator version

```text
openapi-generator-cli 5.2.0
  commit : 90f7bcd
  built  : 2021-07-09T09:41:46Z
  source : https://github.com/openapitools/openapi-generator
  docs   : https://openapi-generator.tech/
```

Not sure if this worked at some point.

##### OpenAPI declaration file content or url

A minimal example can be seen here: <https://gist.github.com/alehed/e5eb44e14f0dcf38a430421e62f9aa12>

##### Generation Details

```bash
openapi-generator-cli generate -i https://gist.githubusercontent.com/alehed/e5eb44e14f0dcf38a430421e62f9aa12/raw/e57d418eeebc8f3553a815522401047c7cebec5a/swapi-dummy.yaml -g elm -o generated
```

##### Steps to reproduce

Run the above command and inspect the generated folder.

**Comment by wing328** [comment]
> openapi-generator-cli 5.2.0

Can you try the latest master? I saw some merged PRs related to elm but not sure if it addresses the issue you reported.

**Comment by alehed** [comment]
I tried with `docker pull openapitools/openapi-generator-cli:latest` not sure if latest always follows master though.

**Comment by dlkj** [comment]
I've also hit this issue but got a bit further digging into what is happening.

If the first model, when sorted alphabetically by name, is an alias of a simple type (`CodegenModel.isAlias == true`) then `DefaultGenerator#generateAllModels` will suppress the generation of `Data.elm`

https://github.com/OpenAPITools/openapi-generator/blob/9170e1030f4c10b460d30d507db4e8996a87b23a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L562-L569

**Comment by wing328** [comment]
@dlkj thanks for the details.

If we skip the isAlias check in line 265 for Elm client generator, would that fix the issue?

**Comment by dlkj** [comment]
I agree with the intention of the logic, I don't think we want to generate models for aliases.

My opinion is that the line:
https://github.com/OpenAPITools/openapi-generator/blob/9170e1030f4c10b460d30d507db4e8996a87b23a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L562

should be changed to filter the model list, removing any alias models.

**Comment by wing328** [comment]
Can you please file a PR with the suggested change? Much easier to review that way.
